bot/handlers/categories_handler.py

```diff
@@ -83,23 +85,61 @@ def change_user_categories(update: Update, context: CallbackContext):
     telegram_id = update.effective_user.id
 
     user_db.change_user_category(telegram_id=telegram_id, category_id=category_id)
-    choose_category(update, context)
+    choose_category(update, context, category_id=category_id)
     update.callback_query.answer()
 
 
+def is_subcategory(category_id):
+    if not category_id:
+        return
+    category = Category.query.options(load_only('id')).filter_by(archive=False).filter_by(id=category_id).first()
+    return True if category.parent_id else False
+
+
+def list_subcategories(category_id):
+    if not category_id:
+        return
+    if is_subcategory(category_id):
+        subcategory_parent_id = Category.query.options(load_only('id')).filter_by(archive=False).filter_by(id=category_id).first().parent_id
```

Comment:
Давай немного отформатируем, сделай что-то подобное, чтобы легче читать было.
```suggestion
        subcategory_parent_id = Category.query.options(load_only('id'))
              .filter_by(archive=False)
              .filter_by(id=category_id)
              .first().parent_id
```

А может быть так, что значения не будет и всё упадёт? Наверно, только если категорию удалили, но тут ничего не поделаешь

Reply:
Насколько я знаю, категорию можно удалить "заархивировав" её, но тогда не будет такой кнопки у бота. Если во время отрисовки бота, удалить руками категорию, тогда тут точно ничего не поделаешь.